WireAuthentication/Sources/WireAuthentication/Components/DetermineAuthMethodComponent.swift

```diff
@@ -38,36 +37,62 @@ protocol DetermineAuthMethodComponentDependency: Dependency {
 
 class DetermineAuthMethodComponent: Component<DetermineAuthMethodComponentDependency> {
 
-    @MainActor private var viewModel: DetermineAuthMethodViewModel {
-        DetermineAuthMethodViewModel(
-            router: dependency.router,
-            factory: self,
-            environmentType: dependency.environmentType,
-            backendConfig: dependency.backendConfig
-        )
+    private let environmentType: BackendEnvironmentType
+    private let backendConfig: BackendConfig
+    private let backendMetadata: BackendMetadata?
```

Comment:
question: I understand these are injected now to accommodate the possibility now that we have received a different backend config. I wonder though, if it makes sense to have a different component `DetermineAuthMethodOnPremComponent` so these changes could be isolated to the new component only. Having said that though, we could keep using the same view and view model. What do you think?

Reply:
I've added `DetermineAuthMethodOnPremComponent`